--- a/drivers/video/fbdev/core/fbcon.c
+++ b/drivers/video/fbdev/core/fbcon.c
@@ -604,6 +604,7 @@
 	if (fb_get_color_depth(&info->var, &info->fix) == 1)
 		erase &= ~0x400;
 	logo_height = fb_prepare_logo(info, ops->rotate);
+	logo_height += (info->var.yres / 2) - (logo_height / 2);
 	logo_lines = DIV_ROUND_UP(logo_height, vc->vc_font.height);
 	q = (unsigned short *) (vc->vc_origin +
 				vc->vc_size_row * rows);
@@ -1071,8 +1072,6 @@
 
 	info = fbcon_info_from_console(vc->vc_num);
 
-	if (logo_shown < 0 && console_loglevel <= CONSOLE_LOGLEVEL_QUIET)
-		logo_shown = FBCON_LOGO_DONTSHOW;
 
 	if (vc != svc || logo_shown == FBCON_LOGO_DONTSHOW ||
 	    (info->fix.type == FB_TYPE_TEXT))
--- a/drivers/video/fbdev/core/fbmem.c
+++ b/drivers/video/fbdev/core/fbmem.c
@@ -55,7 +55,7 @@
 
 bool fb_center_logo __read_mostly;
 
-int fb_logo_count __read_mostly = -1;
+int fb_logo_count __read_mostly = 1;
 
 struct fb_info *get_fb_info(unsigned int idx)
 {
@@ -536,6 +536,7 @@
 		fb_set_logo(info, logo, logo_new, fb_logo.depth);
 	}
 
+	fb_center_logo = true;
 	if (fb_center_logo) {
 		int xres = info->var.xres;
 		int yres = info->var.yres;
